sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -432,25 +441,27 @@ impl InputSelection {
 
                     let account_input = input_accounts
                         .iter()
-                        .find(|i| {
-                            if let Output::Account(account_input) = &i.output {
-                                *account_output.account_id() == account_input.account_id_non_null(i.output_id())
-                            } else {
-                                false
-                            }
-                        })
+                        .find(|i| is_account_with_id(&i.output, account_output.account_id(), i.output_id()))
                         .expect("ISA is broken because there is no account input");
 
-                    if let Err(err) = AccountOutput::transition_inner(
-                        account_input.output.as_account(),
-                        account_output,
-                        &input_chains_foundries,
-                        &self.outputs,
-                    ) {
-                        log::debug!("validate_transitions error {err:?}");
-                        return Err(Error::UnfulfillableRequirement(Requirement::Account(
-                            *account_output.account_id(),
-                        )));
+                    match &account_input.output {
+                        Output::Account(account) => {
+                            if let Err(err) = AccountOutput::transition_inner(
+                                account,
+                                account_output,
+                                &input_chains_foundries,
+                                &self.outputs,
+                            ) {
+                                log::debug!("validate_transitions error {err:?}");
+                                return Err(Error::UnfulfillableRequirement(Requirement::Account(
+                                    *account_output.account_id(),
+                                )));
+                            }
+                        }
+                        Output::Basic(_) => {
+                            // TODO https://github.com/iotaledger/iota-sdk/issues/1664
+                        }
+                        _ => unreachable!(),
```

Comment:
I think Chloé's preference here would be to use `panic` instead since this execution branch hasn't been ruled out prior. It may be unreachable protocol-wise, but not programatically/logically.

Reply:
This is 100% programatically unreachable

Reply:
Or maybe I don't understand what you mean by programatically

Reply:
like ... could it become suddenly reachable by a protocol change?

Reply:
I guess, a massive protocol change that would break the whole codebase. I don't think that's a valid argument anyway, any change to the match could make an unreachable reachable 🤷🏻

Reply:
I'll change it anyway as I don't really care but this is definitely unreachable

Reply:
Changed